src/components/TestToolMenu.js

```diff
@@ -31,7 +29,7 @@ const propTypes = {
 
 const defaultProps = {
     user: {
-        shouldUseStagingServer: false,
+        shouldUseStagingServer: undefined,
```

Comment:
Providing a default value interferes with the `loadasGet` call, where we provide a default value 

If there is a default boolean value, then `lodashGet` would never use the fallback we provide

Maybe this got unnoticed because of how `withOnyx` works, but IMO the plan is to use the Onyx value
And fallback to something (specific) when Onyx does not have a value

The specific thing we want to fallback to is whether we use the staging api right now or not
Unfortunately this cannot be evaluated at the time we declare `defaultProps`

Reply:
>Unfortunately this cannot be evaluated at the time we declare defaultProps

Meaning we can't make the defaultProps use `ApiUtils.isUsingStagingApi()`? That seems fine. 

**alternate idea (non blocking)**: Possibly could be solved by passing a default value for `user.shouldUseStagingServer` to Onyx during initialization and making it a required prop here. Then we could eliminate the `lodashGet` calls as well.

Reply:
> Meaning we can't make the defaultProps use `ApiUtils.isUsingStagingApi()`? That seems fine.

Yes

Reply:
> **alternate idea (non blocking)**: Possibly could be solved by passing a default value for `user.shouldUseStagingServer` to Onyx during initialization and making it a required prop here. Then we could eliminate the `lodashGet` calls as well.

I think this would be a bit complicated

The default value `shouldUseStagingServer` is environment specific 
We need to first get the ENV and then call `Onyx.init` with default value for `user` - `{ shouldUseStagingServer: ENV == staging }` 

I can expose a `ApiUtils.getDefaultApi`, but I can't guarantee it would have the correct value at the time `Onyx.init` is called (similar issues as to why default props value was removed)

Reply:
Maybe I missed it. But I thought the `user.shouldUseStagingServer` value is straight from Onyx and represents the toggle state? My suggestion is to just enforce that it be false by default at the Onyx level.

Reply:
I think it would be fine to do a follow-up. I must admit, I find some of the changes a bit confusing. I don't mean to criticize, as I understand the goal is to make things better. However, I want to make sure that the added complexity is worth it. When you explained it on Slack, it seemed simpler to me.

Reply:
There are 3 options here 
- keep this change, because it's technically correct - when `user` prop is not passed the `shouldUseStagingServer` value should not default to `false`, but `undefined` so that `ApiUtils.isUsingStagingApi()` can be used as fallback 
- discard the change - currently we always have a `user` object passed when this component renders, the object lacks a `shouldUseStagingServer` so it's same as if it's `undefined` (when we haven't touched the toggle, like after fresh login)
- remove `defaultProps` - a `user` is always passed. Avoids the confusion around `defaultProps` with `undefined` value

Reply:
Let's start by leaving a comment explaining why `shouldUseStagingServer` needs to be `undefined`. If after reading that comment everyone decides it is reasonable then we can skip discussing whether it feels too complex.

Reply:
Done

Reply:
The comment makes sense to me ✅